src/main/java/edu/kit/ipd/crowdcontrol/objectservice/quality/QualityIdentificator.java

```diff
@@ -207,18 +207,46 @@ private void rateQualityOfRatings(AnswerRecord answerRecord, Map<AlgorithmRating
      * @param answerRecord  which is going to be rated
      * @param params Mapping of parameter-records to the user specified parameters represented as a string
      */
-    private void rateQualityOfAnswers(AnswerRecord answerRecord, Map<AlgorithmAnswerQualityParamRecord, String> params) {
+    private void rateQualityOfAnswers(ExperimentRecord experimentRecord,AnswerRecord answerRecord, Map<AlgorithmAnswerQualityParamRecord, String> params) {
         Map<String, Integer> result;
         result = answerIdentifier.identifyAnswerQuality(answerRatingOperations, answerRecord, params, MAXIMUM_QUALITY, MINIMUM_QUALITY);
         answerRatingOperations.setQualityToAnswer(answerRecord, result.get(AnswerQualityStrategy.QUALITY));
 
         // Checks if quality_assured bit can be set.
-        if ((((double) result.get(AnswerQualityStrategy.NUM_OF_RATINGS) / (double) experimentOperations.getExperiment(answerRecord.getExperiment())
-                .orElseThrow(() -> new IllegalArgumentException("Error, can't find experiment of given experiment-ID: "+answerRecord.getExperiment())).getRatingsPerAnswer()) >= 0.8)) {
+        if(hasQualityAssured(experimentRecord,answerRecord,result.get(AnswerQualityStrategy.NUM_OF_RATINGS))){
             answerRatingOperations.setAnswerQualityAssured(answerRecord);
         }
 
     }
+
+
+    /**
+     * Checks if the quality of the given answer cannot change anymore, even if it receives more ratings
+     * @param experimentRecord the answers experiment
+     * @param answerRecord the answer
+     * @param numberOfRatings number of good ratings for the answer
+     * @return true if quality of the answer cannot change anymore
+     */
+    private boolean hasQualityAssured(ExperimentRecord experimentRecord,AnswerRecord answerRecord, int numberOfRatings) {
+        int desiredRatingsPerAnswerOfExperiment = experimentOperations.getExperiment(answerRecord.getExperiment())
+                .orElseThrow(() -> new IllegalArgumentException("Error, can't find experiment of given experiment-ID: " + answerRecord.getExperiment())).getRatingsPerAnswer();
+    if(numberOfRatings >= desiredRatingsPerAnswerOfExperiment){
+        return true;
+    }else if((double) numberOfRatings / (double)desiredRatingsPerAnswerOfExperiment >= 0.8){
```

Comment:
it would be possible to set the assured bit earlier than 80% of the required votes.